GNUmakefile

```diff
@@ -1,8 +1,8 @@
-CC=cc
-CFLAGS=-O3 -flto -Wall -g -Werror=implicit-function-declaration -Werror=int-conversion
-LDLIBS=-lm
-INSTALL=install
-PREFIX=/usr
+CC?=cc
+CFLAGS?=-O3 -flto=auto -Wall -g -Werror=implicit-function-declaration -Werror=int-conversion
+LDLIBS?=-lm
+INSTALL?=install
+PREFIX?=/usr
```

Comment:
Is this really needed in another make versions?
Here, GNU make, variables defined in the command line override those in the makefile, see https://www.gnu.org/software/make/manual/html_node/Overriding.html

Reply:
It's important for some of the other operating systems because they pass CC and LDFLAGS and similar via the environment and not the command line (depending on what compiler is installed and default).  For example, AIX and Solaris don't ship any cc in the default PATH, and these systems generally use environment-modules (https://modules.readthedocs.io/) or lmod (https://lmod.readthedocs.io/) for defining these or switching between different installed configurations.

So without the patch you'd need to `gmake CC="$CC" CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS" ...` etc etc each time to override the values, so it's much better to respect the environment IMO.

Reply:
ok!

Another option would be to use `make -e` but I see the benefit of this.

Reply:
Yeah … on Solaris at least, if you set the PATH alone you are mostly OK since their compiler suite provides symlinks for the standard “cc” and friends, but IBM doesn’t on AIX.